sql/src/main/java/io/crate/metadata/Schemas.java

```diff
@@ -102,9 +100,9 @@ public DocTableInfo getDroppableTable(TableIdent tableIdent) {
 
     public DocTableInfo getWritableTable(TableIdent tableIdent) {
         TableInfo tableInfo = getTableInfo(tableIdent);
-        if (!(tableInfo instanceof DocTableInfo)) {
+        if (Operation.isReadOnly(tableInfo)) {
             throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
-                "The table %s is read-only. Write, Drop or Alter operations are not supported", tableInfo.ident()));
+                "The table %s is read-only or an alias. Write, Drop or Alter operations are not supported", tableInfo.ident()));
```

Comment:
Aliases are marked as having only the read only operations. Refactoring the error message seemed more sensible than catching it here, then seeing if the tableInfo is of type DocTableInfo, and if so seeing if it is an alias.